src/pages/LoginPage.jsx

```diff
@@ -1,18 +1,33 @@
 import React from 'react';
+import firebase from 'firebase/app';
+import 'firebase/auth';
 import PageTitle from '../atoms/PageTitle';
-import store from '../Redux/store';
-import LoginChange from '../Redux/actions/LoginChange';
+import TextInputRow from '../atoms/TextInputRow';
+import SubmitButton from '../atoms/SubmitButton';
 
 function LoginPage() {
-  function handleLogin() {
-    store.dispatch(LoginChange(true));
-    window.location.hash = '/armies/view';
+  const [userInfo, setUserInfo] = React.useState({ Email: '', Password: '', Error: '' });
+
+  function handleEmailInput(input) { setUserInfo({ ...userInfo, Email: input }); }
```

Comment:
Good idea to keep functions outside of the render area (as I believe they get re-rendered and garbage collected constantly sometimes). Not a deal breaker, just goo to keep functions a level higher if you can.

Reply:
The functions handleEmailInput, handlePasswordInput, and userLogin are between the state variables and the return function.  If I move them out to a higher level (like outside the LoginPage()) then the state variables are no longer defined.  I guess I don't know what you mean.

Reply:
Well, you can do a refactor on the code block and pull it out with all of those details. State variables are commonly passed in to functions higher up. Not a must do here though, more of a consideration if you need to clean up an area of code.